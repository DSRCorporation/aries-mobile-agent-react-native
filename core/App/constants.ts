import { PredicateType } from '@aries-framework/core'

import { ProofRequestTemplate, ProofRequestType } from './types/proof-reqeust-template'
import { PINValidationRules } from './types/security'

export const defaultLanguage = 'en'
export const dateIntFormat = 'YYYYMMDD'

const lengthOfhiddenAttributes = 10
const unicodeForBulletCharacter = '\u2022'
export const hiddenFieldValue = Array(lengthOfhiddenAttributes).fill(unicodeForBulletCharacter).join('')
// Used to property prefix TestIDs so they can be looked up
// by on-device automated testing systems like SauceLabs.
export const testIdPrefix = 'com.ariesbifold:id/'

export enum LocalStorageKeys {
  Onboarding = 'OnboardingState',
  LoginAttempts = 'LoginAttempts',
  // FIXME: Once hooks are updated this should no longer be necessary
  RevokedCredentials = 'RevokedCredentials',
  RevokedCredentialsMessageDismissed = 'RevokedCredentialsMessageDismissed',
  Preferences = 'PreferencesState',
}

export enum KeychainServices {
  Salt = 'secret.wallet.salt',
  Key = 'secret.wallet.key',
}

export enum EventTypes {
  ERROR_ADDED = 'ErrorAdded',
  ERROR_REMOVED = 'ErrorRemoved',
  BIOMETRY_UPDATE = 'BiometryUpdate',
  BIOMETRY_ERROR = 'BiometryError',
}

export const second = 1000
export const minute = 60000
export const hour = 3600000

// wallet timeout of 5 minutes, 0 means the wallet never locks due to inactivity
export const walletTimeout = minute * 5

/* lockout attempt rules: The base rules apply the lockout at a specified number of incorrect attempts,
 and the threshold rules apply the lockout penalty to each attempt after the threshold that falls on the attemptIncrement.
 (In this case the threshold rule applies to every 5th incorrect login after 20)
5 incorrect => 1 minute lockout
10 incorrect => 10 minute lockout
15 incorrect => 1 hour lockout
20, 25, 30, etc incorrect => 1 day lockout
*/
export const attemptLockoutBaseRules: Record<number, number | undefined> = {
  5: minute,
  10: 10 * minute,
  15: hour,
}
export const attemptLockoutThresholdRules = {
  attemptThreshold: 20,
  attemptIncrement: 5,
  attemptPenalty: 24 * hour,
}

export const walletId = 'walletId'

export const minPINLength = 6

export const PINRules: PINValidationRules = {
  only_numbers: true,
  min_length: 6,
  max_length: 6,
  no_repeated_numbers: false,
  no_repetition_of_the_two_same_numbers: false,
  no_series_of_numbers: false,
  no_even_or_odd_series_of_numbers: false,
  no_cross_pattern: false,
}

export const defaultProofRequestTemplates: Array<ProofRequestTemplate> = [
  {
    id: '8a83675e-f864-4e5a-9c4d-9787f1034c04',
    title: 'Full name',
    details: 'Verify the full name of a person',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedAttributes: [
            {
              name: 'given_names',
            },
            {
              name: 'family_name',
            },
          ],
        },
      ],
    },
  },
  {
    id: '3339c1e1-681f-433c-9f2f-8d0d1af0b3d2',
    title: '19+ and Full name',
    details: 'Verify if a person is 19 years end up and full name.',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedAttributes: [
            {
              name: 'given_names',
            },
            {
              name: 'family_name',
            },
          ],
          requestedPredicates: [
            {
              name: 'age',
              predicateType: PredicateType.GreaterThanOrEqualTo,
              predicateValue: 19,
            },
          ],
        },
      ],
    },
  },
  {
    id: '18183e86-4528-4d10-80f8-aa5c7cbbbfe7',
    title: 'Over 19 years of age',
    details: 'Verify if a person is 19 years end up.',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedPredicates: [
            {
              name: 'age',
              predicateType: PredicateType.GreaterThanOrEqualTo,
              predicateValue: 19,
            },
          ],
        },
      ],
    },
  },
  {
    id: 'd91c7aab-af1a-42b6-8f1f-2702ed6e1f98',
    title: 'Practising lawyer',
    details: 'Verify if a person`is a practicing lawyer.',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Practising Lawyer',
          requestedAttributes: [
            {
              names: ['given_names', 'family_name', 'ppid', 'practicing_status'],
            },
          ],
        },
      ],
    },
  },
  {
    id: 'e4cf6f7c-5abe-4e0d-b299-f944fe5ef8b2',
    title: 'Practising lawyer and full name',
    details: 'Verify if a person`is a practicing lawyer using two different credentials for extra assuarnce',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedAttributes: [
            {
              name: 'given_names',
            },
            {
              name: 'family_name',
            },
          ],
        },
        {
          schema: 'Practising Lawyer',
          requestedAttributes: [
            {
              names: ['given_names', 'family_name', 'ppid', 'practicing_status'],
            },
          ],
        },
      ],
    },
  },
  {
    id: '18183e86-4528-4d10-80f8-aa5c7cbbbfe7',
    title: 'Over some years of age',
    details: 'Verify if a person is over some years ends up.',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedPredicates: [
            {
              name: 'age',
              predicateType: PredicateType.GreaterThanOrEqualTo,
              predicateValue: 19,
              parameterizable: true,
            },
          ],
        },
      ],
    },
  },
]
