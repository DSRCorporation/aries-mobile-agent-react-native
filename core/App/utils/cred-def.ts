import { CredentialMetadataKeys, CredentialExchangeRecord as CredentialRecord } from '@aries-framework/core'

import { parseSchema, parseSchemaFromId, credentialSchema } from './schema'

export function parseCredDefFromId(credDefId?: string, schemaId?: string): string {
  let name = 'Credential'
  if (credDefId) {
    const credDefRegex = /[^:]+/g
    const credDefIdParts = credDefId.match(credDefRegex)
    if (credDefIdParts?.length === 5) {
      name = `${credDefIdParts?.[4].replace(/_|-/g, ' ')}`
        .split(' ')
        .map((credIdPart) => credIdPart.charAt(0).toUpperCase() + credIdPart.substring(1))
        .join(' ')
    }
  }
  if (name.toLocaleLowerCase() === 'default' || name.toLowerCase() === 'credential') {
    name = parseSchemaFromId(schemaId).name
  }
  return name
}

function credentialDefinition(credential: CredentialRecord): string | undefined {
  return credential.metadata.get(CredentialMetadataKeys.IndyCredential)?.credentialDefinitionId
}

export function parsedCredDefNameFromCredential(credential: CredentialRecord): string {
  return parseCredDefFromId(credentialDefinition(credential), credentialSchema(credential))
}

export function parsedCredDefName(credentialDefinitionId: string, schemaId: string): string {
  let credDefName = parseCredDefName(credentialDefinitionId)
  // if credDef name is `default` or `credential` use schema name instead
  if (credDefName.toLocaleLowerCase() === 'default' || credDefName.toLowerCase() === 'credential') {
    credDefName = parseSchema(schemaId).name
  }
  return credDefName
}
