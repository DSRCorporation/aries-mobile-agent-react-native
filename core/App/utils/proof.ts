import { IndyProof, IndyProofRequest } from 'indy-sdk-react-native'

export interface MissingAttribute {
  name: string
}

export interface SharedAttribute {
  name: string
  value: string
}

export interface ResolvedPredicate {
  name: string
  predicateType: string
  predicateValue: number
}

export interface ParsedIndyProof {
  sharedAttribute: Array<SharedAttribute>
  sharedAttributeGroups: Array<Array<SharedAttribute>>
  resolvedPredicates: Array<ResolvedPredicate>
  unresolvedAttributes: Array<MissingAttribute>
  unresolvedAttributeGroups: Array<Array<MissingAttribute>>
  unresolvedPredicates: Array<ResolvedPredicate>
}

export const parseIndyProof = (request: IndyProofRequest, proof: IndyProof) => {
  const sharedAttribute: Array<SharedAttribute> = []
  const sharedAttributeGroups: Array<Array<SharedAttribute>> = []
  const resolvedPredicates: Array<ResolvedPredicate> = []
  const unresolvedAttributes: Array<MissingAttribute> = []
  const unresolvedAttributeGroups: Array<Array<MissingAttribute>> = []
  const unresolvedPredicates: Array<ResolvedPredicate> = []

  for (const [referent, requested_attribute] of Object.entries(request.requested_attributes)) {
    if (requested_attribute.name) {
      const shared = proof.requested_proof.revealed_attrs[referent]
      if (shared) {
        sharedAttribute.push({
          name: requested_attribute.name,
          value: shared.raw,
        })
      } else {
        unresolvedAttributes.push({
          name: requested_attribute.name,
        })
      }
    }

    if (requested_attribute.names) {
      const shared = proof.requested_proof.revealed_attr_groups[referent]
      if (shared) {
        sharedAttributeGroups.push(
          Object.entries(shared.values).map(([name, value]) => ({
            name,
            value: value.raw,
          }))
        )
      } else {
        unresolvedAttributeGroups.push(requested_attribute.names.map((name) => ({ name })))
      }
    }
  }

  for (const [referent, requestedPredicate] of Object.entries(request.requested_predicates)) {
    const shared = proof.requested_proof.requested_predicates[referent]
    if (shared) {
      resolvedPredicates.push({
        name: requestedPredicate.name,
        predicateType: requestedPredicate.p_type,
        predicateValue: requestedPredicate.p_value,
      })
    } else {
      unresolvedPredicates.push({
        name: requestedPredicate.name,
        predicateType: requestedPredicate.p_type,
        predicateValue: requestedPredicate.p_value,
      })
    }
  }

  return {
    sharedAttribute,
    sharedAttributeGroups,
    resolvedPredicates,
    unresolvedAttributes,
    unresolvedAttributeGroups,
    unresolvedPredicates,
  }
}
