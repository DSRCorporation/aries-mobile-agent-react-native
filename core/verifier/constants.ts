import { PredicateType } from '@aries-framework/core'

import { ProofRequestTemplate, ProofRequestType } from './types/proof-reqeust-template'

export const defaultProofRequestTemplates: Array<ProofRequestTemplate> = [
  {
    id: '8a83675e-f864-4e5a-9c4d-9787f1034c04',
    title: 'Full name',
    details: 'Verify the full name of a person',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedAttributes: [
            {
              name: 'given_names',
            },
            {
              name: 'family_name',
            },
            {
              names: ['issued', 'country'],
            },
          ],
          requestedPredicates: [
            {
              name: 'age',
              predicateType: PredicateType.GreaterThanOrEqualTo,
              predicateValue: 17,
            },
          ],
        },
      ],
    },
  },
  {
    id: '3339c1e1-681f-433c-9f2f-8d0d1af0b3d2',
    title: '19+ and Full name',
    details: 'Verify if a person is 19 years end up and full name.',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedAttributes: [
            {
              name: 'given_names',
            },
            {
              name: 'family_name',
            },
          ],
          requestedPredicates: [
            {
              name: 'age',
              predicateType: PredicateType.GreaterThanOrEqualTo,
              predicateValue: 19,
            },
          ],
        },
      ],
    },
  },
  {
    id: '18183e86-4528-4d10-80f8-aa5c7cbbbfe7',
    title: 'Over 19 years of age',
    details: 'Verify if a person is 19 years end up.',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedPredicates: [
            {
              name: 'age',
              predicateType: PredicateType.GreaterThanOrEqualTo,
              predicateValue: 19,
            },
          ],
        },
      ],
    },
  },
  {
    id: 'd91c7aab-af1a-42b6-8f1f-2702ed6e1f98',
    title: 'Practising lawyer',
    details: 'Verify if a person`is a practicing lawyer.',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Practising Lawyer',
          requestedAttributes: [
            {
              names: ['given_names', 'family_name', 'ppid', 'practicing_status'],
            },
          ],
        },
      ],
    },
  },
  {
    id: 'e4cf6f7c-5abe-4e0d-b299-f944fe5ef8b2',
    title: 'Practising lawyer and full name',
    details: 'Verify if a person`is a practicing lawyer using two different credentials for extra assuarnce',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedAttributes: [
            {
              name: 'given_names',
            },
            {
              name: 'family_name',
            },
          ],
        },
        {
          schema: 'Practising Lawyer',
          requestedAttributes: [
            {
              names: ['given_names', 'family_name', 'ppid', 'practicing_status'],
            },
          ],
        },
      ],
    },
  },
  {
    id: '18183e86-4528-4d10-80f8-aa5c7cbbbfe7',
    title: 'Over some years of age',
    details: 'Verify if a person is over some years ends up.',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'Verified Person Schema',
          requestedPredicates: [
            {
              name: 'age',
              predicateType: PredicateType.GreaterThanOrEqualTo,
              predicateValue: 19,
              parameterizable: true,
            },
          ],
        },
      ],
    },
  },
]
