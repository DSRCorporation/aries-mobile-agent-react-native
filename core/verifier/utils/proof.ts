import { Agent, ProofExchangeRecord, ProofIdentifier, ProofState } from '@aries-framework/core'
import { IndyProof, IndyProofRequest } from 'indy-sdk-react-native'

export interface MissingAttribute {
  name: string
}

export interface SharedAttribute {
  name: string
  value: string
  identifiers: ProofIdentifier
}

export interface SharedGroupedAttribute {
  name: string
  value: string
}

export interface SharedAttributesGroup {
  attributes: Array<SharedGroupedAttribute>
  identifiers: ProofIdentifier
}

export interface ResolvedPredicate {
  name: string
  predicateType: string
  predicateValue: number
  identifiers: ProofIdentifier
}

export interface UnresolvedPredicate {
  name: string
  predicateType: string
  predicateValue: number
}

export class ParsedIndyProof {
  public sharedAttributes: Array<SharedAttribute>
  public sharedAttributeGroups: Array<SharedAttributesGroup>
  public resolvedPredicates: Array<ResolvedPredicate>
  public unresolvedAttributes: Array<MissingAttribute>
  public unresolvedAttributeGroups: Array<Array<MissingAttribute>>
  public unresolvedPredicates: Array<UnresolvedPredicate>

  public constructor() {
    this.sharedAttributes = []
    this.sharedAttributeGroups = []
    this.resolvedPredicates = []
    this.unresolvedAttributes = []
    this.unresolvedAttributeGroups = []
    this.unresolvedPredicates = []
  }
}

/*
 * Extract identifiers from indy proof
 * */
const getProofIdentifiers = (proof: IndyProof, proofIndex: number): ProofIdentifier => {
  const identifiers = proof.identifiers[proofIndex]
  if (!identifiers) {
    throw new Error('Invalid indy proof data')
  }
  return {
    schemaId: identifiers.schema_id,
    credentialDefinitionId: identifiers.cred_def_id,
    revocationRegistryId: identifiers.rev_reg_id,
    timestamp: identifiers.timestamp,
  }
}

/*
 * Process indy proof and return
 *  - shared attributes
 *  - shared attribute groups
 *  - resolved predicates
 *  - missing attributes
 *  - missing attribute groups
 *  - unresolved predicates
 * */
export const parseIndyProof = (request: IndyProofRequest, proof: IndyProof): ParsedIndyProof => {
  const result = new ParsedIndyProof()

  for (const [referent, requested_attribute] of Object.entries(request.requested_attributes)) {
    if (requested_attribute.name) {
      const shared = proof.requested_proof.revealed_attrs[referent]
      if (shared) {
        const identifiers = getProofIdentifiers(proof, shared.sub_proof_index)
        result.sharedAttributes.push({
          name: requested_attribute.name,
          value: shared.raw,
          identifiers,
        })
      } else {
        result.unresolvedAttributes.push({
          name: requested_attribute.name,
        })
      }
    }

    if (requested_attribute.names) {
      const shared = proof.requested_proof.revealed_attr_groups[referent]
      if (shared) {
        const attributes = Object.entries(shared.values).map(([name, value]) => ({
          name,
          value: value.raw,
        }))
        const identifiers = getProofIdentifiers(proof, shared.sub_proof_index)
        result.sharedAttributeGroups.push({
          attributes,
          identifiers,
        })
      } else {
        result.unresolvedAttributeGroups.push(requested_attribute.names.map((name) => ({ name })))
      }
    }
  }

  for (const [referent, requestedPredicate] of Object.entries(request.requested_predicates)) {
    // @ts-ignore Mistake in AFJ type definition
    const shared = proof.requested_proof.predicates[referent]
    if (shared) {
      const identifiers = getProofIdentifiers(proof, shared.sub_proof_index)
      result.resolvedPredicates.push({
        name: requestedPredicate.name,
        predicateType: requestedPredicate.p_type,
        predicateValue: requestedPredicate.p_value,
        identifiers,
      })
    } else {
      result.unresolvedPredicates.push({
        name: requestedPredicate.name,
        predicateType: requestedPredicate.p_type,
        predicateValue: requestedPredicate.p_value,
      })
    }
  }

  return result
}

export class CredentialSharedProofData {
  public sharedAttributes: Array<SharedAttribute>
  public sharedAttributeGroups: Array<SharedAttributesGroup>
  public resolvedPredicates: Array<ResolvedPredicate>

  public constructor() {
    this.sharedAttributes = []
    this.sharedAttributeGroups = []
    this.resolvedPredicates = []
  }
}

export type GroupedSharedProofDataItem = { data: CredentialSharedProofData; identifiers: ProofIdentifier }
export type GroupedSharedProofData = Map<string, GroupedSharedProofDataItem>

/*
 * Group parsed indy proof data ( returned from `parseIndyProof`) by credential definition id
 * */
export const groupSharedProofDataByCredential = (data: ParsedIndyProof): GroupedSharedProofData => {
  const result: GroupedSharedProofData = new Map<string, GroupedSharedProofDataItem>()
  for (const item of data.sharedAttributes) {
    if (!result.has(item.identifiers.credentialDefinitionId)) {
      result.set(item.identifiers.credentialDefinitionId, {
        data: new CredentialSharedProofData(),
        identifiers: item.identifiers,
      })
    }
    result.get(item.identifiers.credentialDefinitionId)?.data.sharedAttributes.push(item)
  }
  for (const item of data.sharedAttributeGroups) {
    if (!result.has(item.identifiers.credentialDefinitionId)) {
      result.set(item.identifiers.credentialDefinitionId, {
        data: new CredentialSharedProofData(),
        identifiers: item.identifiers,
      })
    }
    result.get(item.identifiers.credentialDefinitionId)?.data.sharedAttributeGroups.push(item)
  }
  for (const item of data.resolvedPredicates) {
    if (!result.has(item.identifiers.credentialDefinitionId)) {
      result.set(item.identifiers.credentialDefinitionId, {
        data: new CredentialSharedProofData(),
        identifiers: item.identifiers,
      })
    }
    result.get(item.identifiers.credentialDefinitionId)?.data.resolvedPredicates.push(item)
  }
  return result
}

/*
 * Retrieve proof details from AFJ record
 * */
export const getProofData = async (agent: Agent, recordId: string): Promise<ParsedIndyProof | undefined> => {
  const data = await agent.proofs.getFormatData(recordId)
  if (data.request?.indy && data.presentation?.indy) {
    return parseIndyProof(data.request.indy, data.presentation.indy)
  }
  return undefined
}

/*
 * Check if a presentation received
 * */
export const isPresentationReceived = (record: ProofExchangeRecord) => {
  return record.state === ProofState.PresentationReceived || record.state === ProofState.Done
}
