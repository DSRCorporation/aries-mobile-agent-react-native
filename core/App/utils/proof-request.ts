import { defaultProofRequestTemplates } from '../constants'
import {
  IndyRequestedAttribute,
  IndyRequestedPredicate,
  ProofRequestTemplate,
  ProofRequestType,
} from '../types/proof-reqeust-template'

export const buildProofRequestDataByTemplate = (template: ProofRequestTemplate) => {
  if (template.payload.type === ProofRequestType.Indy) {
    const requestedAttributes: Map<string, IndyRequestedAttribute> = new Map()
    const requestedPredicates: Map<string, IndyRequestedPredicate> = new Map()
    let index = 0

    template.payload.data.forEach((data) => {
      if (data.requestedAttributes?.length) {
        data.requestedAttributes.forEach((requestedAttribute) => {
          requestedAttributes.set(`referent_${index}`, requestedAttribute)
          index++
        })
      }
      if (data.requestedPredicates?.length) {
        data.requestedPredicates.forEach((requestedPredicate) => {
          requestedPredicates.set(`referent_${index}`, requestedPredicate)
          index++
        })
      }
    })
    return {
      indy: {
        name: template.title,
        version: template.version,
        nonce: Date.now().toString(),
        requestedAttributes,
        requestedPredicates,
      },
    }
  }
  if (template.payload.type === ProofRequestType.DIF) {
    return {}
  }
}

export const buildProofRequestDataByTemplateId = (id: string) => {
  const template = defaultProofRequestTemplates.find((template) => template.id === id)
  if (!template) {
    return null
  }
  return buildProofRequestDataByTemplate(template)
}
