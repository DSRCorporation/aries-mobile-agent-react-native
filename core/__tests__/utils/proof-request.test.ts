import { defaultProofRequestTemplates } from '../../App/constants'
import { buildProofRequestDataByTemplate } from '../../App/utils/proof-request'
import SpyInstance = jest.SpyInstance

describe('Helpers', () => {
  let spy: SpyInstance

  beforeAll(() => {
    spy = jest.spyOn(Date, 'now').mockImplementation(() => 1677766511505)
  })

  test('Build indy proof request from template containing two requested attributes', async () => {
    const template = defaultProofRequestTemplates[0]
    const proofRequest = buildProofRequestDataByTemplate(template)
    expect(proofRequest).toMatchSnapshot()
  })

  test('Build indy proof request from template containing two requested attributes and predicate', async () => {
    const template = defaultProofRequestTemplates[1]
    const proofRequest = buildProofRequestDataByTemplate(template)
    expect(proofRequest).toMatchSnapshot()
  })

  test('Build indy proof request from template containing attributes group', async () => {
    const template = defaultProofRequestTemplates[3]
    const proofRequest = buildProofRequestDataByTemplate(template)
    expect(proofRequest).toMatchSnapshot()
  })

  afterAll(() => {
    spy.mockRestore()
  })
})
