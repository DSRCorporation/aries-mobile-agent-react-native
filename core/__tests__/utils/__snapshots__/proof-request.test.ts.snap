// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helpers Build indy proof request from template containing attributes group 1`] = `
Object {
  "indy": Object {
    "name": "Practising lawyer",
    "nonce": "1677766511505",
    "requestedAttributes": Map {
      "referent_0" => Object {
        "names": Array [
          "given_names",
          "family_name",
          "ppid",
          "practicing_status",
        ],
      },
    },
    "requestedPredicates": Map {},
    "version": "0.0.1",
  },
}
`;

exports[`Helpers Build indy proof request from template containing two requested attributes 1`] = `
Object {
  "indy": Object {
    "name": "Full name",
    "nonce": "1677766511505",
    "requestedAttributes": Map {
      "referent_0" => Object {
        "name": "given_names",
      },
      "referent_1" => Object {
        "name": "family_name",
      },
    },
    "requestedPredicates": Map {},
    "version": "0.0.1",
  },
}
`;

exports[`Helpers Build indy proof request from template containing two requested attributes and predicate 1`] = `
Object {
  "indy": Object {
    "name": "19+ and Full name",
    "nonce": "1677766511505",
    "requestedAttributes": Map {
      "referent_0" => Object {
        "name": "given_names",
      },
      "referent_1" => Object {
        "name": "family_name",
      },
    },
    "requestedPredicates": Map {
      "referent_2" => Object {
        "name": "age",
        "predicateType": ">=",
        "predicateValue": 19,
      },
    },
    "version": "0.0.1",
  },
}
`;
